openpype/hosts/batchpublisher/controller.py

```diff
@@ -0,0 +1,376 @@
+        # file_mappings = project_settings["batchpublisher"].get(
+        #     "file_mappings", [])
+        file_mappings = GLOB_SEARCH_TO_PRODUCT_INFO_MAP
+        for file_mapping in file_mappings:
+            product_type = file_mapping["product_type"]
+            glob_full_path = directory + "/" + file_mapping["glob"]
+            representation_name = file_mapping.get("representation_name")
+            files = glob.glob(glob_full_path, recursive=False)
+            for filepath in files:
+                filename = os.path.basename(filepath)
+                frame_start = None
+                frame_end = None
+                if filename.count(".") >= 2:
+                    # Lets add the star in place of the frame number
+                    filepath_parts = filepath.split(".")
+                    filepath_parts[-2] = "#" * len(filepath_parts[-2])
+                    # Replace the file path with the version with star in it
+                    filepath = ".".join(filepath_parts)
+                    frames = self._get_frames_for_filepath(filepath)
+                    frame_start = frames[0]
+                    frame_end = frames[-1]
+                # Do not add ingest file path, if it's already been added
+                if filepath in product_items:
+                    continue
+                _filename_no_ext, extension = os.path.splitext(filename)
+                # Create representation name from extension
+                representation_name = representation_name or \
+                    EXT_TO_REP_NAME_MAP.get(extension)
+                if not representation_name:
+                    representation_name = extension.lstrip(".")
+                product_item = ProductItem(
+                    filepath,
+                    product_type,
+                    representation_name,
+                    frame_start=frame_start,
+                    frame_end=frame_end)
+                product_items[filepath] = product_item
+
+        # Walk the entire directory structure again
+        # and look for product items to add.
+        # This time we are looking for product types
+        # by PRODUCT_TYPE_TO_EXT_MAP
+        for root, _dirs, filenames in os.walk(directory, topdown=False):
+            for filename in filenames:
+                filepath = os.path.join(root, filename)
+                filename = os.path.basename(filepath)
+                # Get frame infomration (if any)
+                frame_start = None
+                frame_end = None
+                if filename.count(".") >= 2:
+                    # Lets add the star in place of the frame number
+                    filepath_parts = filepath.split(".")
+                    filepath_parts[-2] = "*"
+                    # Replace the file path with the version with star in it
+                    filepath = ".".join(filepath_parts)
+                    frames = self._get_frames_for_filepath(filepath)
```

Comment:
this will crash in case you can split the file but it isn't sequence - or rather there the part supposed to be frame number actually isn't:

```
Traceback (most recent call last):
  File "C:\Users\annat\Documents\Projects\Ayon\OpenPype\openpype\hosts\batchpublisher\ui\window.py", line 129, in _on_dir_change
    self._table_view.set_current_directory(directory)
  File "C:\Users\annat\Documents\Projects\Ayon\OpenPype\openpype\hosts\batchpublisher\ui\batch_publisher_view.py", line 47, in set_current_directory
    self._model.set_current_directory(directory)
  File "C:\Users\annat\Documents\Projects\Ayon\OpenPype\openpype\hosts\batchpublisher\ui\batch_publisher_model.py", line 34, in set_current_directory
    self._populate_from_directory(directory)
  File "C:\Users\annat\Documents\Projects\Ayon\OpenPype\openpype\hosts\batchpublisher\ui\batch_publisher_model.py", line 198, in _populate_from_directory    self._product_items = self._controller.get_product_items(
  File "C:\Users\annat\Documents\Projects\Ayon\OpenPype\openpype\hosts\batchpublisher\controller.py", line 281, in get_product_items
    frames = self._get_frames_for_filepath(filepath)
  File "C:\Users\annat\Documents\Projects\Ayon\OpenPype\openpype\hosts\batchpublisher\controller.py", line 374, in _get_frames_for_filepath
    frame = int(filepath_parts[-2])
ValueError: invalid literal for int() with base 10: 'ma'
```

And I know, I ran it on totally random folder :)